src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilterTest.java

```diff
@@ -469,21 +470,30 @@ public void testSuppressById() throws Exception {
     public void testTagsAreClearedEachRun() {
         final SuppressWithNearbyCommentFilter suppressionCommentFilter =
                 new SuppressWithNearbyCommentFilter();
-        final FileContents contents =
-                new FileContents("filename", "//SUPPRESS CHECKSTYLE ignore", "line2");
+        final List<?> tags1 = accept(suppressionCommentFilter,
+                "filename1", "//SUPPRESS CHECKSTYLE ignore this");
+        assertEquals("Invalid tags size", 1, tags1.size());
+        final List<?> tags2 = accept(suppressionCommentFilter,
+                "filename2", "No comments in this file");
+        assertEquals("Invalid tags size", 0, tags2.size());
+    }
+
+    /**
+     * Calls the filter with a minimal set of inputs and returns a list of
+     * {@link SuppressWithNearbyCommentFilter} internal type {@code Tag}.
+     * Testing implementation details is a bad practice, but we also need 100% test coverage.
```

Comment:
> Testing implementation details is a bad practice

I do not get this ... testing is always good. Please remove this phrase.

Reply:
Well, this is my personal preference. I believe that the black-box testing is absolutely necessary, while the white-box testing hampers refactoring. Fixed.